### PR TITLE
Feat!(duckdb): add support for median function of duckdb to postgres

### PR DESCRIPTION
**Pull request by longxiaofei**

### CODE REVIEW

File: tests/dialects/test_duckdb.py [hunk 1]
```diff
@@ -359,6 +359,16 @@ def test_duckdb(self):
 
         self.validate_identity("SELECT ISNAN(x)")
 
+        self.validate_all(
+            "MEDIAN(amount)",
+            write={
+                "postgres": "PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY amount)",
+                "presto": "APPROX_PERCENTILE(amount, 0.5)",
+                "spark": "PERCENTILE_APPROX(amount, 0.5)",
+                "duckdb": "APPROX_QUANTILE(amount, 0.5)",
```
Comment (GeorgeSittas) [inline tests/dialects/test_duckdb.py:368]: DuckDB's [docs](https://duckdb.org/docs/sql/aggregates) mention that the formula for `MEDIAN(x)` is `QUANTILE_CONT(x, 0.5)`; is the transformation to `APPROX_QUANTILE(x, 0.5)` lossless / equivalent to the `CONT` variant? Are `NULL` values handled differently?

File: tests/dialects/test_postgres.py [hunk 2]
```diff
@@ -212,6 +212,15 @@ def test_postgres(self):
...
         )
+        self.validate_all(
+            "SELECT PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY amount)",
+            read={
+                "databricks": "SELECT PERCENTILE_APPROX(amount, 0.5)",
+                "presto": "SELECT APPROX_PERCENTILE(amount, 0.5)",
+                "spark": "SELECT PERCENTILE_APPROX(amount, 0.5)",
+                "trino": "SELECT APPROX_PERCENTILE(amount, 0.5)",
+            },
+        )
```
Comment (GeorgeSittas) [inline tests/dialects/test_postgres.py:223]: Does this transpile correctly to DuckDB too? If not, is it easy to fix by e.g. including the `remove_within_group_for_percentiles` transformation for DuckDB?

**Review by GeorgeSittas — COMMENTED** [review]
Thanks for the PR - left a couple of comments / questions.

I think this should be marked as breaking? (i.e. `Feat` -> `Feat!`)

**Comment by longxiaofei** [comment]
Thanks for reviews, I will modify it later.

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -4717,6 +4717,14 @@ class Quantile(AggFunc):
     arg_types = {"this": True, "quantile": True}
 
 
+class QuantileCont(Quantile):
+    _sql_names = ["QUANTILE_CONT"]
+
+
+class QuantileDisc(Quantile):
```
Comment (tobymao) [inline sqlglot/expressions.py:4724]: somu

**Comment by GeorgeSittas** [comment]
Closing this in favor of #2219.

**Pull request closed without merging**